src/main/scala/io/iohk/ethereum/network/p2p/messages/CommonMessages.scala

```diff
@@ -12,45 +12,131 @@ import org.bouncycastle.util.encoders.Hex
 
 object CommonMessages {
   object Status {
-    val code: Int = Versions.SubProtocolOffset + 0x00
+    // TODO: shouldn't we rather just use `protocolVersion` to distinguish these messages?
+    val code63: Int = Versions.SubProtocolOffset + 0x00
+    val code64: Int = Versions.SubProtocolOffset + 0x11
 
     implicit class StatusEnc(val underlyingMsg: Status)
         extends MessageSerializableImplicit[Status](underlyingMsg)
         with RLPSerializable {
-      override def code: Int = Status.code
+      override def code: Int = underlyingMsg.code
 
       override def toRLPEncodable: RLPEncodeable = {
         import msg._
-        RLPList(protocolVersion, networkId, totalDifficulty, bestHash, genesisHash)
+        msg match {
+          case _: Status63 =>
+            RLPList(protocolVersion, networkId, totalDifficulty, bestHash, genesisHash)
+
+          case _: Status64 =>
+            RLPList(protocolVersion, networkId, totalDifficulty, latestCheckpointNumber, bestHash, genesisHash)
+        }
       }
     }
 
     implicit class StatusDec(val bytes: Array[Byte]) extends AnyVal {
-      def toStatus: Status = rawDecode(bytes) match {
-        case RLPList(protocolVersion, networkId, totalDifficulty, bestHash, genesisHash) =>
-          Status(protocolVersion, networkId, totalDifficulty, bestHash, genesisHash)
+      def toStatus(code: Int): Status = (code, rawDecode(bytes)) match {
+        case (
+              `code63`,
+              RLPList(
+                protocolVersion,
+                networkId,
+                totalDifficulty,
+                bestHash,
+                genesisHash
+              )
+            ) =>
+          Status63(protocolVersion, networkId, totalDifficulty, bestHash, genesisHash)
+
+        case (
+              `code64`,
+              RLPList(
+                protocolVersion,
+                networkId,
+                totalDifficulty,
+                latestCheckpointNumber,
```

Comment:
would it be better to have `latestCheckpointNumber` as the last field on list, as it's something which might not exist?

Reply:
Not sure, does it matter?

Reply:
not now, but IMHO as encoding to the RLP means encode to the list of things it should be easier to decode it having optional fields on the end than having them somewhere in the middle

Reply:
Well, it's not a case of optionality here. These are different messages with different codes, ie. once the code is determined a completely different structure may follow.
It's not a big deal the order doesn't really matter but these fields are closely related, in fact, after ETCM-263 they will correspond to a single type - `ChainWeight`, perhaps then it should even be encoded as a nested RLPList? 🤔